components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1236,33 +1238,56 @@ void RewardsServiceImpl::LoadURL(
     return;
   }
 
-  net::URLFetcher::RequestType request_type = URLMethodToRequestType(method);
-
-  net::URLFetcher* fetcher = net::URLFetcher::Create(
-      parsed_url, request_type, this).release();
-  fetcher->SetRequestContext(g_browser_process->system_request_context());
-  fetcher->SetAutomaticallyRetryOnNetworkChanges(kRetriesCountOnNetworkChange);
-
+  if (test_response_callback_) {
+    std::string test_response;
+    std::map<std::string, std::string> test_headers;
+    test_response_callback_.Run(url, &test_response, &test_headers),
+    callback(200, test_response, test_headers);
+    return;
+  }
+
+  net::NetworkTrafficAnnotationTag traffic_annotation =
```

Comment:
@NejcZdovc Need help to fill this annotation. (I don't have any background about rewards service.)